talk/objectorientation/operators.tex

```diff
@@ -86,6 +86,32 @@
   \end{block}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitlecpp[98]{Friend declarations}
+  \begin{block}{Concept}
+    \begin{itemize}
+      \item Functions/classes can be declared \mintinline{cpp}{friend} within a class scope
+      \item They gain access to all private/protected members
+      \item Useful for operators such as \mintinline{cpp}{a + b}
+      \item Don't abuse friends to go around a wrongly designed interface
+      \item Avoid unexpected modifications of class state in a friend
+    \end{itemize}
+  \end{block}
+  \begin{block}{\texttt{operator+} as a friend}
+    \small
+    \begin{cppcode*}{gobble=2}
+      class Complex {
+        float m_real, m_imag;
```

Comment:
This does not compile if you do not declare a constructor
```cpp
Complex( float, float );
```

Reply:
Fixed in #229

Reply:
@hageboeck originally had ` // + Complex(float, float) ...` on that line as a comment, and I suggested to make `Complex` a `struct`. We must have missed this in the end.